### ISSUE TITLE
More granular test results

### ISSUE THREAD

**Issue by ljharb** [getify/es-feature-tests#6]
`ArrayMethods` isn't particularly useful to me. However, an exact report of which methods passed and failed would be wonderful.

The design I think would be amazing is (using "supports" as the root report object):
`supports.Array.prototype = true` = all tested `Array.prototype` methods pass
`supports.Array.prototype.fill = true` = `Array.prototype.fill` passes
`supports.Array` = all tested `Array` methods pass
`supports.Array.of` = `Array.of` passes
… and so forth.

For everything that's an API method, this would be very clean. You could then do `supports.syntax.arrowFunctions` etc if there was concern about namespace overlapping. Alternatively, the above could be `supports.api.Array` etc.

(filed per https://twitter.com/getify/status/596773738277785600)

**Comment by getify** [comment]
Thanks! Will come back and post some thoughts ASAP. :)

**Comment by Mouvedia** [comment]
Just to be sure you want `supports.A` to only report the support at the first level?
If `supports.A.*` are supported but `supports.A.B.C` isn't, would `supports.A` be true?

**Comment by ljharb** [comment]
@Mouvedia no, I'd want `supports.A` passing to indicate that every single test nested in that namespace passed.
